chaincfg/chainec/edwards.go

```diff
@@ -278,7 +279,7 @@ func newEdwardsDSA() DSA {
 			return edwards.GenerateKey(edwardsCurve, rand)
 		},
 		sign: func(priv PrivateKey, hash []byte) (r, s *big.Int, err error) {
-			if priv.GetType() != ECTypeEdwards {
+			if priv.GetType() != dcrec.ECTypeEdwards {
```

Comment:
`edwardsDSA` is a wrapper for the underlining type. I think we're better off removing this and using the underlining type directly.